### PR TITLE
`azurerm_windows_web_app` : fix the issue of incorrect application stack setting after updating the web app without changing the `application_stack` configuration

### PR DESCRIPTION
**Pull request by sinbai**
Two purposes of this PR:

- Application stack is incorrectly updated

 Symptom: 

> 1. Creating `azurerm_windows_web_app` with `site_config` as shown below.
> 
>   ```
> site_config {
> 	health_check_path           = "/health"
> 	load_balancing_mode="LeastRequests"
>     application_stack {
>         current_stack                = "java"
> 	    java_embedded_server_enabled = true
>         java_version                 = "11"
>       } 
> }
> ```
> 2. Update `azurerm_windows_web_app` and only change `health_check_eviction_time_in_min ` from null to 5 as follows.
>  
> ```
> site_config {
>         health_check_eviction_time_in_min = 5
> 	health_check_path           = "/health"
> 	load_balancing_mode="LeastRequests"
>         application_stack {
>            current_stack                = "java"
> 	    java_embedded_server_enabled = true
>             java_version                 = "11"
>         }
> }
> ```
> 
> Actual:
> image
> Expected:
> image
> 
> Per the following code in the [existing](https://github.com/hashicorp/terraform-provider-azurerm/blob/6f8b827b654f0145d7f9f39b57b8784abb2b5df5/internal/services/appservice/windows_web_app_resource.go#L511C4-L511C4) in Read(), I assume this should also be added to Update() function to solve the above issue.
> 
> ```
> // Despite being part of the defined `Get` response model, site_config is always nil so we get it explicitly
> webAppSiteConfig, err := client.GetConfiguration(ctx, id.ResourceGroup, id.SiteName)
> if err != nil {
>    return fmt.Errorf("reading Site Config for Windows %s: %+v", id, err)
> }
> ```

- I assume that we should only set `JavaVersion` in Read() function when `currentStack` is `java`  since the API will return the value of `JavaVersion` that was once set.

Test result:
PASS: TestAccWindowsWebApp_updateAppStack (611.57s)

### CODE REVIEW

**Review by jackofallops — CHANGES_REQUESTED** [review]
Hi @sinbai - Thanks for this change and apologies for the delayed review as we've been discussing this internally trying to find a better way to work with the app_stack values for Windows apps. I've left one comment for an additional change, since the `current_stack` property is `Optional` we should not rely on it. Given that there is other reliance on the value already present, we're going to accept it, but we'll need the comment adding to the docs for now, and we'll be taking a look at addressing it appropriately for 4.0 as we believe it will require a breaking change in the longer term.
Thanks!

File: internal/services/appservice/helpers/windows_web_app_schema.go [hunk 1]
```diff
@@ -817,7 +817,11 @@ func (s *SiteConfigWindows) Flatten(appSiteConfig *web.SiteConfig, currentStack
 	}
 	winAppStack.PythonVersion = pointer.From(appSiteConfig.PythonVersion) // This _should_ always be `""`
 	winAppStack.Python = currentStack == CurrentStackPython
-	winAppStack.JavaVersion = pointer.From(appSiteConfig.JavaVersion)
+
+	// we should only set JavaVersion when  currentStack is java since the API will return the value of JavaVersion that was once set
+	if currentStack == "java" {
```
[the remark this answers is not in the document]
Comment (sinbai) [reply internal/services/appservice/helpers/windows_web_app_schema.go:822]: Hi @jackofallops thanks for your feedback. The doc has been updated. Could you please take another look?

**Review by jackofallops — COMMENTED** [review]
Hi @sinbai - Thanks for adding the requested change.  During testing I've found that this is actually doesn't work as expected due to a slightly larger problem / bug. I'm investigating now and I may need to push some changes to your branch to fix the broader problem. If that happens, I'll get another maintainer to review my additions and we'll get this in as soon as it's all good and testing works as expected.

**Review by tombuildsstuff — APPROVED** [review]
LGTM 👍

**Pull request merged**